stdlib/public/core/KeyPath.swift

```diff
@@ -1731,40 +1807,78 @@ internal struct RawKeyPathComponent {
       // 'modify' access.
       Builtin.performInstantaneousReadAccess(offsetAddress._rawValue,
         NewValue.self)
-      return .continue(offsetAddress
-        .assumingMemoryBound(to: NewValue.self)
-        .pointee)
+
+      pointer.initialize(
+        to: offsetAddress.assumingMemoryBound(to: NewValue.self).pointee
+      )
 
     case .get(id: _, accessors: let accessors, argument: let argument),
          .mutatingGetSet(id: _, accessors: let accessors, argument: let argument),
          .nonmutatingGetSet(id: _, accessors: let accessors, argument: let argument):
-      return .continue(accessors.getter()(base,
-                               argument?.data.baseAddress ?? accessors._value,
-                               argument?.data.count ?? 0))
+      let getter: ComputedAccessorsPtr.Getter<CurValue, NewValue> = accessors.getter()
+
+      pointer.initialize(
+        to: getter(
+          base,
+          argument?.data.baseAddress ?? accessors._value,
+          argument?.data.count ?? 0
+        )
+      )
 
     case .optionalChain:
       _internalInvariant(CurValue.self == Optional<NewValue>.self,
                    "should be unwrapping optional value")
       _internalInvariant(_isOptional(LeafValue.self),
                    "leaf result should be optional")
-      if let baseValue = unsafeBitCast(base, to: Optional<NewValue>.self) {
-        return .continue(baseValue)
-      } else {
-        // TODO: A more efficient way of getting the `none` representation
-        // of a dynamically-optional type...
-        return .break((Optional<()>.none as Any) as! LeafValue)
+
+      // Optional's tags are some = 0, none = 1
+      let tag = UInt32(Builtin.getEnumTag(base))
+
+      if _fastPath(tag == 0) {
+        // Optional "shares" a layout with its Wrapped type meaning we can
+        // reinterpret the base address as an address to its Wrapped value.
+        pointer.initialize(to: Builtin.reinterpretCast(base))
```

Comment:
Where do we ensure the tag byte, if any, after the base value is zero?

Reply:
Not sure I understand what you mean here 🤔

Reply:
If `Optional<T>` is larger than `T` (because T doesn't have any extra inhabitants) then the tag byte needs to be zero to indicate the `some` case.

Reply:
Oh I check that right before this:

```swift
      // Optional's tags are some = 0, none = 1
       let tag = UInt32(Builtin.getEnumTag(base))
```
I call the vwt to get the enum tag of the value and if it's 0 then I know I can bit cast the optional value to its wrapped one.